src/data-model-providers/codegen/CodegenDataModelProvider.cpp

```diff
@@ -366,20 +366,19 @@ CHIP_ERROR CodegenDataModelProvider::Attributes(const ConcreteClusterPath & path
         ReturnErrorOnFailure(builder.Append(AttributeEntryFrom(path, attribute)));
     }
 
-    // This "GlobalListEntry" is specific for metadata that ember does not include
-    // in its attribute list metadata.
-    //
-    // By spec these Attribute/AcceptedCommands/GeneratedCommants lists are:
-    //   - lists of elements
-    //   - read-only, with read privilege view
-    //   - fixed value (no such flag exists, so this is not a quality flag we set/track)
-    DataModel::AttributeEntry globalListEntry(0, /* default initial value, attributeId is assigned later */
-                                              DataModel::AttributeQualityFlags::kListAttribute, Access::Privilege::kView,
-                                              std::nullopt);
-
     for (auto & attribute : GlobalAttributesNotInMetadata)
     {
-        globalListEntry.attributeId = attribute;
+
+        // This "GlobalListEntry" is specific for metadata that ember does not include
+        // in its attribute list metadata.
+        //
+        // By spec these Attribute/AcceptedCommands/GeneratedCommants lists are:
+        //   - lists of elements
+        //   - read-only, with read privilege view
+        //   - fixed value (no such flag exists, so this is not a quality flag we set/track)
+        DataModel::AttributeEntry globalListEntry(attribute, DataModel::AttributeQualityFlags::kListAttribute,
+                                                  Access::Privilege::kView, std::nullopt);
+
         ReturnErrorOnFailure(builder.Append(globalListEntry));
```

Comment:
maybe this works a bit faster?
```suggestion
        DataModel::AttributeEntry globalListEntry(attribute, DataModel::AttributeQualityFlags::kListAttribute,
                                                  Access::Privilege::kView, std::nullopt);

        ReturnErrorOnFailure(builder.Append(std::move(globalListEntry)));
```

or even this:

```suggestion
        ReturnErrorOnFailure(builder.Append(DataModel::AttributeEntry{
            attribute, 
            DataModel::AttributeQualityFlags::kListAttribute,
            Access::Privilege::kView, std::nullopt
            }));
```

Reply:
Yes, the `std:move` probably will save copy operations after the instantiation.

But the second suggestion is even better, since the object would be created in situ, the appended. Let me test this.

Reply:
`std::move()` most probably improves performance by skipping unnecessary `copy` calls by using **move semantics**. It also reduces the codesize of the binaries.

The call of the constructor inside `builder.Append()` increases the codesize.

Implementing first suggestion, `std:move()`.